modules/exoplanets/tasks/exoplanetsdatapreparationtask.cpp

```diff
@@ -146,37 +130,91 @@ void ExoplanetsDataPreparationTask::perform(
     inputDataFile.clear();
     inputDataFile.seekg(0);
 
-    // Read past the first line, containing the data names
-    readFirstDataRow(inputDataFile, row);
+    // The reading is restarted, so we need to read past the first line,
+    // containing the data names, again
+    readFirstDataRow(inputDataFile);
 
     LINFO(fmt::format("Loading {} exoplanets", total));
 
+    int exoplanetCount = 0;
+    while (std::getline(inputDataFile, row)) {
+        ++exoplanetCount;
+        progressCallback(static_cast<float>(exoplanetCount) / static_cast<float>(total));
+
+        PlanetData planetData = parseDataRow(
+            row,
+            columnNames,
+            _inputSpeckPath,
+            _teffToBvFilePath
+        );
+
+        // Create look-up table
+        long pos = static_cast<long>(binFile.tellp());
+        std::string planetName = planetData.host + " " + planetData.component;
+        lutFile << planetName << "," << pos << std::endl;
+
+        binFile.write(reinterpret_cast<char*>(&(planetData.dataEntry)), sizeof(ExoplanetDataEntry));
+    }
+
+    progressCallback(1.f);
+}
+
+std::vector<std::string>
+ExoplanetsDataPreparationTask::readFirstDataRow(std::ifstream& file)
+{
+    std::string line;
+
+    // Read past any comments and empty lines
+    while (std::getline(file, line)) {
+        bool shouldSkip = line[0] == '#' || line.empty();
```

Comment:
```suggestion
        bool shouldSkip = line.empty() || line[0] == '#';
```

Otherwise we try to look at the first character of a line that is empty